core/lib/snapshots_applier/src/tests.rs

```diff
@@ -0,0 +1,229 @@
+use std::collections::HashMap;
+
+use async_trait::async_trait;
+use zksync_types::{api::en::SyncBlock, snapshots::SnapshotHeader, MiniblockNumber};
+use zksync_web3_decl::jsonrpsee::core::ClientError as RpcError;
+
+use crate::SnapshotsApplierMainNodeClient;
+
+#[derive(Debug, Default)]
+struct MockMainNodeClient {
+    fetch_l2_block_responses: HashMap<MiniblockNumber, SyncBlock>,
+    fetch_newest_snapshot_response: Option<SnapshotHeader>,
+}
+
+#[async_trait]
+impl SnapshotsApplierMainNodeClient for MockMainNodeClient {
+    async fn fetch_l2_block(&self, number: MiniblockNumber) -> Result<Option<SyncBlock>, RpcError> {
+        if let Some(response) = self.fetch_l2_block_responses.get(&number) {
+            Ok(Some((*response).clone()))
+        } else {
+            Ok(None)
+        }
+    }
+
+    async fn fetch_newest_snapshot(&self) -> Result<Option<SnapshotHeader>, RpcError> {
+        if let Some(response) = self.fetch_newest_snapshot_response.clone() {
+            Ok(Some(response))
+        } else {
+            Ok(None)
+        }
+    }
+}
+#[cfg(test)]
+mod snapshots_applier_tests {
```

Comment:
Bikeshedding: this module looks redundant. I'd rather move utils above in `tests::utils` module.

Reply:
which methods exactly do you suggest moving to test utils?